test/compilable/test9274.d

```diff
@@ -0,0 +1,23 @@
+// https://issues.dlang.org/show_bug.cgi?id=9274
+struct S
+{
+    float[] arr;
+    alias arr this;
+}
+
+static assert(!is(S == float[])); // ok
+static assert(!is(S == T[], T)); // fails
```

Comment:
It's probably testes elsewhere, but I'd like to see positive tests next to the negative ones for completeness:
```D
static assert(is(S : float[]));
static assert(is(S : T[], T));
```

Reply:
Added.